### DISCUSSION TITLE
I cant send Template Messages

### DISCUSSION THREAD

**Discussion by kingmariano** [tulir/whatsmeow#569] [WhatsApp protocol Q&A]
`func (cfg *waConfig) SendMessage(ctx context.Context, client *whatsmeow.Client, chatJID , senderJID types.JID, message, username, messageType string) error {
	switch messageType {
	case "templateMessage":
		TemplateMessage := &waProto.TemplateMessage{
			HydratedTemplate: &waProto.TemplateMessage_HydratedFourRowTemplate{
				Title: &waProto.TemplateMessage_HydratedFourRowTemplate_HydratedTitleText{
					HydratedTitleText: "The Title",
				  },
				  TemplateId:          proto.String("template-id"),
				  HydratedContentText: proto.String("The Content"),
				  HydratedFooterText:  proto.String("The Footer"),
				//   MaskLinkedDevices: proto.Bool(true),
				  HydratedButtons: []*waProto.HydratedTemplateButton{
					 // This for URL button
					 {
						Index: proto.Uint32(1),
						HydratedButton: &waProto.HydratedTemplateButton_UrlButton{
						  UrlButton: &waProto.HydratedTemplateButton_HydratedURLButton{
							DisplayText: proto.String("The Link"),
							Url:         proto.String("https://fb.me/this"),
						  },
						},
					  },
					   // This for call button
					   {
						Index: proto.Uint32(2),
						HydratedButton: &waProto.HydratedTemplateButton_CallButton{
						  CallButton: &waProto.HydratedTemplateButton_HydratedCallButton{
							DisplayText: proto.String("Call us"),
							PhoneNumber: proto.String("1234567890"),
						  },
						},
					  },
					  {
						Index: proto.Uint32(3),
						HydratedButton: &waProto.HydratedTemplateButton_QuickReplyButton{
						  QuickReplyButton: &waProto.HydratedTemplateButton_HydratedQuickReplyButton{
							DisplayText: proto.String("Quick reply"),
							Id:          proto.String("quick-id"),
						  },
						},
					  },
				  },
			},
			
		}
          log.Print("four_row_template",TemplateMessage.GetHydratedTemplate() ,"---\n")
		  log.Print("templateid", TemplateMessage.TemplateId,"---\n")
		  log.Print("templateString",TemplateMessage.String(),"---\n")
		TemplateMessage.ProtoMessage()
		
		res, err := client.SendMessage(ctx, senderJID, &waProto.Message{
			TemplateMessage: TemplateMessage,
		}, whatsmeow.SendRequestExtra{
			Peer: true,
		})
		if err != nil {
			return err
		}
		fmt.Print(res.Timestamp, res.ID)
		resendMSG := client.BuildUnavailableMessageRequest(chatJID, senderJID, res.ID)
		resend, err := client.SendMessage(ctx, senderJID, &waProto.Message{
			TemplateMessage: resendMSG.TemplateMessage,
		}, whatsmeow.SendRequestExtra{
			Peer: true,
		})
		if err != nil {
			return err
		}
		fmt.Print(resend.Timestamp, resend.ID)
		
	case "conversation":
		res, err := client.SendMessage(ctx, senderJID, &waProto.Message{
			Conversation: proto.String(FormatMessage(message, username)),
		}, whatsmeow.SendRequestExtra{
			Peer: false,
		})
		if err != nil {
			return err
		}
		fmt.Println(res.ID, res.Timestamp)
	default:
		log.Fatal("not a message type")
	}
	return nil
}`

[Client WARN] Server returned different participant list hash when sending to 2347036179840@s.whatsapp.net. Some devices may not have received the message.
Yet i am still getting this error

**Comment by kingmariano** [comment]
I cant send any other message except conversation. please help

**Comment by hrizal** [comment]
It's been a long time since messages in template form can no longer be sent
(they can be sent, but the message will never reach the destination number)
this may be due to restrictions from WhatsApp side
Forget about sending messages in the form of buttons or templates

Pada Sab, 13 Apr 2024 pukul 07.27 Charles Ozochukwu <
***@***.***> menulis:

**Reply by fcavalcantirj** [reply]
how to send messages with like YES/NO button please?

**Reply by ghost** [reply]
You can probably put the latest sent messageID.

```go

questionID := ""

// initiating the question
resp, err := client.SendMessage(questionMsg ....)
if err != nil {
    panic(err)
}

questionID = rep.ID
```

The listening part is to check if the message `questionID` is being quoted by checking the qoutedID from the quoted message.
